ivadomed/loader/slice_filter.py

```diff
@@ -4,65 +4,101 @@
 
 
 class SliceFilter(object):
-    """Filter 2D slices from dataset.
+    """Filter 2D slices or patches from dataset.
 
-    If a sample does not meet certain conditions, it is discarded from the dataset.
+    If a slice or 2D patch does not meet certain conditions, it is discarded from the dataset.
 
     Args:
-        filter_empty_mask (bool): If True, samples where all voxel labels are zeros are discarded.
-        filter_empty_input (bool): If True, samples where all voxel intensities are zeros are discarded.
-        filter_absent_class (bool): If True, samples where all voxel labels are zero for one or more classes are discarded.
-        filter_classification (bool): If True, samples where all images fail a custom classifier filter are discarded.
+        filter_empty_mask (bool): If True, slices where all voxel labels are zeros are discarded.
+        filter_absent_class (bool): If True, slices where all voxel labels are zero for one or more classes are discarded.
+        filter_empty_input (bool): If True, slices where all voxel intensities are zeros are discarded.
+        filter_classification (bool): If True, slices where all images fail a custom classifier filter are discarded.
+        filter_empty_mask_patch (bool): If True, 2D patches where all voxel labels are zeros are discarded at training time.
+        filter_absent_class_patch (bool): If True, 2D patches where all voxel labels are zero for one or more classes are discarded at training time.
+        filter_empty_input_patch (bool): If True, 2D patches where all voxel intensities are zeros are discarded at training time.
+        device (torch.device): Indicates the CPU or GPU ID.
+        cuda_available (bool): If True, CUDA is available.
+        is_train (bool): Indicates if at training time.
 
     Attributes:
-        filter_empty_mask (bool): If True, samples where all voxel labels are zeros are discarded.
-        filter_empty_input (bool): If True, samples where all voxel intensities are zeros are discarded.
-        filter_absent_class (bool): If True, samples where all voxel labels are zero for one or more classes are discarded.
-        filter_classification (bool): If True, samples where all images fail a custom classifier filter are discarded.
+        filter_empty_mask (bool): If True, slices where all voxel labels are zeros are discarded.
+        filter_absent_class (bool): If True, slices where all voxel labels are zero for one or more classes are discarded.
+        filter_empty_input (bool): If True, slices where all voxel intensities are zeros are discarded.
+        filter_classification (bool): If True, slices where all images fail a custom classifier filter are discarded.
```

Comment:
Just wondering if it is "all" images or "any" images? The code seems to suggest that as soon as any image failed the classifier, it will return false?

Reply:
Thanks for the explanation! Still not sure when that is really used or if we have a test case for it but seems like a very edge feature. Your explanation makes sense.